### PR TITLE
external-workloads: Run cilium-cli inside a container

### PR DESCRIPTION
**Pull request by michi-covalent**
Update externalworklads.yaml to run cilium-cli inside a container instead of using cilium-cli-test-job-chart. Update docker run command to specify --user flag so that install-external-workload.sh gets created with correct permissions.

Ref: #2623
Ref: #2627
Ref: cilium/design-cfps#9

### CODE REVIEW

**Comment by michi-covalent** [comment]
✅ success: https://github.com/cilium/cilium-cli/actions/runs/9671104454/job/26681073692?pr=2630

marking it ready for review

**Review by viktor-kurchenko — APPROVED** [review]
Cool!

File: .github/workflows/externalworkloads.yaml [hunk 1]
```diff
@@ -147,55 +147,52 @@ jobs:
...
+            --set kubeProxyReplacement=true \
+            --set loadBalancer.l7.backend=envoy \
+            --set tls.secretsBackend=k8s \
+            --set ipv4NativeRoutingCIDR="${{ steps.cluster.outputs.cluster_cidr }}"
+
+          # Enable Relay
+          cilium hubble enable
+
+          # Wait for cilium and hubble relay to be ready
+          # NB: necessary to work against occassional flakes due to https://github.com/cilium/cilium-cli/issues/918
```
Comment (viktor-kurchenko) [inline .github/workflows/externalworkloads.yaml:174]: NIT: `occassional` -> `occasional`.

Comment (michi-covalent) [reply .github/workflows/externalworkloads.yaml]: fixed! 🔧

**Pull request merged**